Countries Info/ViewControllers/CountryViewCell.swift

```diff
@@ -8,34 +8,55 @@
 import UIKit
 
 class CountryViewCell: UICollectionViewCell {
-
+    
     @IBOutlet weak var countryNameLabel: UILabel!
     @IBOutlet weak var capitalLabel: UILabel!
     @IBOutlet weak var regionLabel: UILabel!
     
     @IBOutlet weak var flagImageView: UIImageView!
     @IBOutlet weak var countryStack: UIStackView!
     @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
-
+    
     var viewModel: CountryCollectionViewCellViewModelProtocol? {
         didSet {
-                countryNameLabel.text = self.viewModel?.countryName
-                capitalLabel.text = self.viewModel?.capital
-                regionLabel.text = self.viewModel?.region
-                
-                if self.viewModel != nil {
-                    activityIndicator.stopAnimating()
-                    countryStack.isHidden = false
-                    flagImageView.isHidden = false
-                }  
+            configure()
         }
     }
-
+    
     override func awakeFromNib() {
         super.awakeFromNib()
         setUI()
     }
     
+    private func configure() {
+        countryNameLabel.text = viewModel?.countryName
+        capitalLabel.text = viewModel?.capital
+        regionLabel.text = viewModel?.region
+        DispatchQueue.global().async {
+            self.imageGenerator(code: self.viewModel?.alpha2Code) { (data) in
+                DispatchQueue.main.async {
+                    if let imageData = data {
+                        self.flagImageView.image = UIImage(data: imageData)
+                    } else {
+                        self.flagImageView.image = UIImage(named: NameConstants.unknownFlagImage)
+                    }
+                    self.activityIndicator.stopAnimating()
+                    self.countryStack.isHidden = false
+                    self.flagImageView.isHidden = false
+                }
+            }
+        }
+    }
+    
+    private func imageGenerator(code: String?, completion: @escaping (Data?) -> Void) {
```

Comment:
We have the same function with the same idea on the Details View Controller, maybe let put this func under some factory-like **image factory** to didn't copy the same function.